lang/csharp/src/apache/benchmark/Avro.benchmark.csproj

```diff
@@ -31,6 +31,12 @@
     <IsPackable>false</IsPackable>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <!-- Some schemas use lower cased names, which causes some class names being lower case only -->
+    <!-- e.g. The type name 'test' only contains lower-cased ascii characters. Such names may become reserved for the language. -->
+    <NoWarn>$(NoWarn);CS8981</NoWarn>
+  </PropertyGroup>
```

Comment:
This disables that warning only in the becnchmark project. There was only 1 warning  for https://github.com/apache/avro/blob/master/lang/csharp/src/apache/benchmark/org/apache/avro/benchmark/small/test.cs#L33. Since that file was originally generated by avrogen, I did not wanto to change the actual class name or disable the warning in that specific file.

All other projects will generate that warning.

Reply:
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/warning-waves

btw I think using `@identifier` is not related to this, since that is regarding for existing keyword conflicts. This warning si simply about not using lower case names for types to avoid any potential future C# keyword conflict.

Reply:
Using @ in front of an all lower case type name could be a avrogen option in the future for sure and would hide this warning for sure. However this is true for pretty much for all languages, that new keywords could cause breaking changes.

Reply:
Currently there is unit test in the library which makes sure that the reserved keyword list is up to date with the Roslyn compiler's keyword list. This means that if a new keyword is added, a unit test will fail which indicates that the keyword list must be updated. Once tthat keyword list is updated, avrogen will add @ to a type name if it is a newly introduced keyword.

https://github.com/apache/avro/blob/master/lang/csharp/src/apache/test/CodGen/CodeGenTest.cs#L37-L55
#1595